src/textmate/src/grammar.ts

```diff
@@ -0,0 +1,412 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import * as tm from "@azure-tools/tmlanguage-generator";
+import path from "path";
+import plist from "plist";
+
+export const grammarPath = path.resolve(__dirname, '../bicep.tmlanguage');
+
+type IncludeRule = tm.IncludeRule<BicepScope>;
+type BeginEndRule = tm.BeginEndRule<BicepScope>;
+type MatchRule = tm.MatchRule<BicepScope>;
+type Grammar = tm.Grammar<BicepScope>;
+
+type BicepScope =
+  | "comment.block.bicep"
+  | "comment.line.double-slash.bicep"
+  | "constant.character.escape.bicep"
+  | "constant.numeric.bicep"
+  | "constant.language.bicep"
+  | "entity.name.type.bicep"
+  | "entity.name.function.bicep"
+  | "keyword.control.declaration.bicep"
+  | "keyword.other.bicep"
+  | "string.quoted.single.bicep"
+  | "string.quoted.multi.bicep"
+  | "variable.name.bicep"
+  | "variable.other.readwrite.bicep"
+  | "variable.other.property.bicep"
+  | "meta.objectliteral.bicep"
+  | "meta.object.member.bicep"
+  | "punctuation.definition.template-expression.begin.bicep" 
+  | "punctuation.definition.template-expression.end.bicep";
+
+const bounded = (text: string) => `\\b${text}\\b`;
+const after = (regex: string) => `(?<=${regex})`;
+const notAfter = (regex: string) => `(?<!${regex})`;
+const before = (regex: string) => `(?=${regex})`;
+const notBefore = (regex: string) => `(?!${regex})`;
+
+const meta: typeof tm.meta = tm.meta;
+const identifierStart = "[_$[:alpha:]]";
+const identifierContinue = "[_$[:alnum:]]";
+const identifier = bounded(`${identifierStart}${identifierContinue}*`);
+
+// whitespace. ideally we'd tokenize in-line block comments, but that's a lot of work. For now, ignore them.
+const ws = `(?:\\s|/\\*.*\\*/)*`;
+
+const lineComment: MatchRule = {
+  key: "line-comment",
+  scope: "comment.line.double-slash.bicep",
+  match: `//.*$`,
+};
+
+const blockComment: BeginEndRule = {
+  key: "block-comment",
+  scope: "comment.block.bicep",
+  begin: `/\\*`,
+  end: `\\*/`,
+};
+
+const comments: IncludeRule = {
+  key: 'comments',
+  patterns: [lineComment, blockComment],
+};
+
+const expression: IncludeRule = {
+  key: "expression",
+  patterns: [
+    /* placeholder filled later due to cycle*/
+  ],
+};
+
+const escapeChar: MatchRule = {
+  key: "escape-character",
+  scope: "constant.character.escape.bicep",
+  match: `\\\\(u[0-9A-Fa-f]{6}|n|r|t|\\\\|'|\\\${)`,
```

Comment:
>6 [](start = 28, length = 1)

We support a variable number of digits in the unicode codepoint IIRC.

Reply:
Good point, corrected.